### PR TITLE
NVSHAS-6572: Parse image config spec referred in manifest

### PR DESCRIPTION
**Pull request by becitsthere**

### CODE REVIEW

File: share/scan/registry.go [hunk 1]
```diff
@@ -70,52 +70,79 @@ func IsQuayRegistry(rc *RegClient) bool {
...
-func BuildV2ImageInfo(dg string, body []byte) (imageInfo ImageInfo, parsedSchemaVersion int, err error) {
+func (rc *RegClient) copyV2LayerAndHistory(imageInfo *ImageInfo, manV2 *manifestV2.Manifest, ccmi *registry.ManifestInfo) bool {
+	allLayersAreCosignPayloads := true
+
+	// In the history list from container image config spec, only the layer that has no empty_layer flag
+	// has a digest in the manifest layer list.
+	// The following section bring the layer list in imageInfo to the same size as history (cmd)
+	j := len(manV2.Layers) - 1
+	for i := 0; i < len(ccmi.Cmds); i++ {
+		if (ccmi != nil && ccmi.EmptyLayers[i]) || j < 0 {
```
Comment (williamlin-suse) [inline share/scan/registry.go:81]: ccmi should be always non-nil when it reaches line 81 or it would panic in line 80

Comment (becitsthere) [reply share/scan/registry.go:81]: ok

File: share/scan/registry.go [hunk 2]
```diff
@@ -70,52 +70,79 @@ func IsQuayRegistry(rc *RegClient) bool {
...
 
-func BuildV2ImageInfo(dg string, body []byte) (imageInfo ImageInfo, parsedSchemaVersion int, err error) {
+func (rc *RegClient) copyV2LayerAndHistory(imageInfo *ImageInfo, manV2 *manifestV2.Manifest, ccmi *registry.ManifestInfo) bool {
+	allLayersAreCosignPayloads := true
+
+	// In the history list from container image config spec, only the layer that has no empty_layer flag
+	// has a digest in the manifest layer list.
+	// The following section bring the layer list in imageInfo to the same size as history (cmd)
+	j := len(manV2.Layers) - 1
+	for i := 0; i < len(ccmi.Cmds); i++ {
```
Comment (williamlin-suse) [inline share/scan/registry.go:80]: it looks like ccmi could be nil when copyV2LayerAndHistory() is called

File: share/scan/registry.go [hunk 3]
```diff
@@ -70,52 +70,79 @@ func IsQuayRegistry(rc *RegClient) bool {
 	return strings.EqualFold(rc.URL[:len(quayRegistryURL)], quayRegistryURL)
 }
 
-func BuildV2ImageInfo(dg string, body []byte) (imageInfo ImageInfo, parsedSchemaVersion int, err error) {
+func (rc *RegClient) copyV2LayerAndHistory(imageInfo *ImageInfo, manV2 *manifestV2.Manifest, ccmi *registry.ManifestInfo) bool {
+	allLayersAreCosignPayloads := true
+
```
Comment (williamlin-suse) [inline share/scan/registry.go:75]: Should imageInfo.Layers be reset to empty first after entering this function?
Or imageInfo.Layers is always empty when copyV2LayerAndHistory() is called?

File: share/scan/registry.go (hunk 2 above)
Comment (becitsthere) [reply share/scan/registry.go:80]: OK

File: share/scan/registry.go (hunk 3 above)
Comment (becitsthere) [reply share/scan/registry.go:75]: It is initiated in GetImageInfo(). It is always empty when entering here.

Comment (becitsthere) [reply share/scan/registry.go:75]: Maybe we should, because there are two places where buildV2ImageInfo() is called.

**Pull request merged**